torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -361,23 +361,13 @@ void RequestCallbackImpl::processRpc(
 
       auto& pythonRpcHandler = PythonRpcHandler::getInstance();
       IValue py_ivalue;
-      try {
-        {
-          py::gil_scoped_acquire acquire;
-          py_ivalue = jit::toIValue(
-              pythonRpcHandler.runPythonUdf(std::move(uprc).movePythonUdf()),
-              PyObjectType::get());
-        }
-        ownerRRef->setValue(std::move(py_ivalue));
-      } catch (py::error_already_set& e) {
-        // py::error_already_set requires GIL to destruct, take special care.
-        ownerRRef->setError(e.what());
+      {
         py::gil_scoped_acquire acquire;
-        e.restore();
-        PyErr_Clear();
-      } catch (std::exception& e) {
-        ownerRRef->setError(e.what());
```

Comment:
[_run_function](https://github.com/pytorch/pytorch/blob/master/torch/distributed/rpc/internal.py#L149) doesn't seem to set the error on the ownerRRef though? Don't we still need this?

Reply:
This was never executed, as exceptions thrown by the python udf is wrapped in a RemoteException object, which will be re-thrown when calling `to_here()`.

Reply:
Hmm, FWIW, those cbs also all have their own try-catch and doesn't seem would throw an error.

Reply:
Let me move this PR out of this stack, as we are unsure about the impact here. And will add some more tests later.